### ISSUE TITLE
Question about the code in the README

### ISSUE THREAD

**Issue by cgiosy** [disjukr/bunja#6]
Hello. In [the example code](https://github.com/disjukr/bunja?tab=readme-ov-file#defining-a-bunja-that-relies-on-other-bunja) of the README, I think there is a possibility that `send("unsubscribe-foo")` may not work because the `socket.close()` part is executed first. Is this the intended behavior?

**Comment by disjukr** [comment]
Thank you for showing interest in bunja.

That's right. As noted, the `cleanup` method for `websocketBunja` will be called first, so if all places where both `websocketBunja` and `resourceFooBunja` are simultaneously relied upon are removed, the `send("unsubscribe-foo")` part won’t hold much significance.

However, in scenarios where `websocketBunja` continues to be used while `resourceFooBunja` is no longer referenced (for instance, as in the example where `PageA` relies on `resourceFooBunja` and `PageB` relies on `resourceBarBunja`—navigating from `PageA` to `PageB` would result in `websocketBunja` still being used while `resourceFooBunja` is cleaned up), the `send("unsubscribe-foo")` code becomes meaningful and therefore needs to remain.

**Comment by cgiosy** [comment]
Okay. I was worried that the child's cleanup would not work properly because it accessed the parent's object, which had already been destroyed when the parent's cleanup was executed.
However, if we assume that the child is already cleaned up when the parent's cleanup is executed, it seems okay because there is no need to access the parent object. Thank you for your answer.

**Comment by disjukr** [comment]
> Okay. I was worried that the child's cleanup would not work properly because it accessed the parent's object, which had already been destroyed when the parent's cleanup was executed.

Yes, your concern is valid. When the child’s cleanup is executed, the parent may have already been cleaned up. This is something that must be taken into account when using Bunja.

A similar discussion has already taken place in React, so you may want to refer to that for additional context: https://github.com/facebook/react/issues/16728

> However, if we assume that the child is already cleaned up when the parent's cleanup is executed, it seems okay because there is no need to access the parent object. Thank you for your answer.

As mentioned earlier, references to both the parent and child can disappear simultaneously, so the situation you are concerned about can still arise. It’s worth thinking about whether there’s any chance of issues arising when writing the code.
